File: Java/ons-webhook-api-test/src/main/java/br/org/ons/ons/sintegre/controller/WebhookController.java
package br.org.ons.ons.sintegre.controller;

import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.Consumes;
import javax.ws.rs.core.Response;
import javax.ws.rs.client.Client;
import javax.ws.rs.client.ClientBuilder;
import br.org.ons.ons.sintegre.model.WebhookRequestModel;

/**
 * Controller para rebimento das notificações oriundas do SINtegre.
 * 
 * @author Jose Mauro da Silva Sandy - Rerum
 * @date 24/07/2019 13:04:41
 */
@Path("/webhook")
public class WebhookController {

    /**
     * Recebe o POST do ONS informando a existência de novas informações para um determinado produto.
     * @param request informações referentes ao produto.
     */
    @POST
    @Consumes("application/json; charset=UTF-8")
    @Produces("application/json; charset=UTF-8")
    public void post(WebhookRequestModel request) {
        // Retornar StatusCode 20X em caso de processamento com sucesso. Qualquer outro código será considerado erro e a mensagem reenviada.
        
        Client client = ClientBuilder.newClient();
        Response response = client.target(request.getUrl()).request().get();
        
        if (response.getStatus() == 200) {
            // Processar salvamento do arquivo.
        } else {
            // Processar retorno de erro.
        }
    }
}

File: C#/BasicAuthentication/ONS.Webhook.Api.Test/Controllers/WebhookController.cs
using System.Net.Http;
using Microsoft.AspNetCore.Mvc;
using ONS.Webhook.Api.Test.Models;
using Microsoft.AspNetCore.Authorization;

namespace ONS.Webhook.Api.Test.Controllers
{
    /// <summary>
    /// Controller para rebimento das notificações oriundas do SINtegre.
    /// </summary>
    /// <Author>Jose Mauro da Silva Sandy - Rerum</Author>
    /// <Date>19/07/2019 11:46:07</Date>
    [Authorize]
    [ApiController]
    [Route("api/[controller]")]
    public partial class WebhookController : ControllerBase
    {
        #region Api Methods

        /// <summary>
        /// Recebe o POST do ONS informando a existência de novas informações para um determinado produto.
        /// </summary>
        /// <param name="request">informações referentes ao produto.</param>
        [HttpPost]
        public async void Post([FromBody] WebhookRequestModel request)
        {
            // Retornar StatusCode 20X em caso de processamento com sucesso. Qualquer outro código será considerado erro e a mensagem reenviada.
            using (var httpClient = new HttpClient())
            {
                var response = await httpClient.GetAsync(request.Url);
                if (response.IsSuccessStatusCode)
                {
                    // Processar salvamento do arquivo.
                }
                else
                {
                    // Processar retorno de erro.
                }
            }
        }

        #endregion  
    }
}
